internal/driver/driver.go

```diff
@@ -129,39 +166,79 @@ func (s *Driver) HandleWriteCommands(deviceName string, protocols map[string]mod
 		deviceLocation = fmt.Sprintf("%v", protocol["deviceLocation"])
 		baudRate, _ = cast.ToIntE(protocol["baudRate"])
 
-		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): protocol = %v, device location = %v, baud rate = %v", i, deviceLocation, baudRate))
+		s.lc.Debugf("Driver.HandleWriteCommands(): protocol = %v, device location = %v, baud rate = %v", i, deviceLocation, baudRate)
 	}
 
 	for i, req := range reqs {
-		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): deviceResourceName = %v", req.DeviceResourceName))
-		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): protocols: %v, resource: %v, attribute: %v, parameters: %v", protocols, req.DeviceResourceName, req.Attributes, params))
+		s.lc.Debugf("Driver.HandleWriteCommands(): deviceResourceName = %v", req.DeviceResourceName)
+		s.lc.Debugf("Driver.HandleWriteCommands(): protocols: %v, resource: %v, attribute: %v, parameters: %v", protocols, req.DeviceResourceName, req.Attributes, params)
 
-		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
+		// Get the value type from device profile
+		valueType := req.Type
+		s.lc.Debugf("Driver.HandleWriteCommands(): value type = %v", valueType)
+
+		var value interface{}
+		var err error
 
+		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
 		if key_type_value == "generic" {
-			if value, err := params[i].StringValue(); err == nil {
-				key_timeout_value, _ := cast.ToIntE(req.Attributes["timeout"])
+			switch valueType {
+			case common.ValueTypeInt8:
+				value, err = params[i].Int8Value()
+			case common.ValueTypeInt16:
+				value, err = params[i].Int16Value()
+			case common.ValueTypeString:
+				value, err = params[i].StringValue()
+			default:
+				return fmt.Errorf("Driver.HandleWriteCommands(): Unsupported value type: %v", valueType)
+			}
+
+			if err != nil {
+				return err
+			} else {
```

Comment:
Remove else to avoid unneeded indentation. Code below doesn't need to be in an else due to the return statement above.